Milestone 3 | query multiple tables

ITurres opened the pull request:
# 🚩I have made the following changes to complete Vet Clinic - Milestone/Project 3

## Here is a summary of what has been done

### Type of **`files`** changed/added

![My Skills](https://skillicons.dev/icons?i=postgresql)

### In this new Branch, I have **✏️MODIFIED✏️** the succeeding **\*📂 folders** and/or **\*\*📄 files** to complete this milestone's [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_multiple_tables.md)

# \*📂 root/

- **\*\*📄 data.sql** ✏️

  - I have inserted data into the `owners` and `species` tables
  - I have also updated the `animals` table with the data related to the `owners` and `species` tables.

- **\*\*📄 queries.sql** ✏️

  - New queries have been made to the `animals`, `owners` and `species` tables.

- **\*\*📄 schema.sql** ✏️

  - I have removed the **species** column and added the following:

    - **species_id** column.
    - **owner_id** column.
    - Two **FOREIGN KEY** constraints to the `species` and `owners` tables.

  - I created the two tables mentioned above:

    - `species` table.
    - `owners` table.

---

### 📸 Screenshots of queries made to the database

  <details>
    <summary>SQL Query Collection- Click to expand!</summary>

![vet-clinic-project-3-queries-123](https://github.com/ITurres/vet-clinic-db/assets/100724715/19276877-5de6-4fe8-a972-593d28989fda)

![vet-clinic-project-3-queries-456](https://github.com/ITurres/vet-clinic-db/assets/100724715/ed8ca37a-e77e-424a-ae25-508e3a05b4a3)

![vet-clinic-project-3-queries-7last](https://github.com/ITurres/vet-clinic-db/assets/100724715/5b7b4b1f-4f58-4d05-8221-55033f816bbc)


  </details>

---

### Thank you for taking the time to review this PR ⭐

---

If you require additional information or have any questions, please contact me on Slack as _**Arturo (Arthur) Emanuel Guerra Iturres**_. I'll be happy to help you out. 🎯

---

ITurres committed:
feat: on 'schema.sql' Modify -animals- and Add new tables

- Modified the -animals- table by:
    - Removing the 'species' column.
    - Adding both 'species_id' and 'owner_id' columns.
    - Adding a foreign key constraint to the 'species_id' column.
    - Adding a foreign key constraint to the 'owner_id' column.

- Added two new tables -species- and -owners-

    - Both with a primary key column named 'id' and
      a column named 'name' for -species- / 'full_name' for -owners-.
    - owners- have 'age' column.

ITurres committed:
feat: on 'data.sql' add data to the tables

- Inserted new data into -owners- and -species- table.
- Modified inserted animals-data so it includes the corresponding 'owner_id' and
  'species_id' value.

ITurres committed:
fix: on 'data.sql' fix CASE on -line 51-52- Insertion

ITurres committed:
feat: on 'queries.sql' Added new queries based on project-3 requirements

bobb-Rob commented on schema.sql:
```diff
@@ -6,5 +6,23 @@ CREATE TABLE
         neutered BOOLEAN NOT NULL,
         weight_kg NUMERIC(10, 2) NOT NULL,
-        species VARCHAR(50) NULL
+        species_id INT,
+        owner_id INT,
+        CONSTRAINT fk_species FOREIGN KEY (species_id) REFERENCES species (id) ON DELETE CASCADE,
+        CONSTRAINT fk_owners FOREIGN KEY (owner_id) REFERENCES owners (id) ON DELETE CASCADE
+    );
```
- [x] When implementing modifications to existing database tables, it is considered a recommended practice to utilize the ALTER keyword for executing new statements that enact these changes. In this context:

- The "species" column is being removed.
- A new column named "species_id" is being introduced, establishing a foreign key relationship with the "species" table.
- An additional column named "owner_id" is being appended, establishing a foreign key relationship with the "owners" table.

The SQL commands for these actions can be structured as follows:

```sql
ALTER TABLE animals DROP COLUMN species;
ALTER TABLE animals ADD COLUMN species_id INT REFERENCES species(id);
ALTER TABLE animals ADD COLUMN owner_id INT REFERENCES owners(id);
```

This approach ensures that alterations to the database schema are carried out in a controlled manner, aligning with best practices for maintaining data integrity and database structure.

Please remove the commands from lines 9 to 12 for the animal table and use the above-suggested method.

ITurres replied:
Hi @bobb-Rob I hope you are well and cheers for reviewing this pr! ⭐️

---

All changes were executed using the **psql** tool:

I utilized the ALTER command to appropriately modify tables for deletion and updating in accordance with the new requirements.

These SQL statements were implemented via **psql** to address the subsequent requirements.

However, the clarity regarding whether these statements should be documented or if direct modifications to the target table were necessary was not provided by the requirements or were ambiguous.

If you require additional details or have any inquiries, please don't hesitate to let me know. 🤠

Davidosky007 replied:
* As mentioned in the [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_multiple_tables.md#:~:text=Remove%20column%20species) you are to remove the species column, currently, your queries hasn't solved this yet. kindly fix this

ITurres replied:
Hi Guys! Good day 🌞

All right then, changes will be made to follow the suggested required changes 🎯 

Cheers again! 😃

bobb-Rob reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️
Hi @ITurres, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
![](https://camo.githubusercontent.com/34482800e1b92cb79ddca06edb4051edc407cb190f896e1e949e596aa9073851/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)

## Highlights
✅ Data insertion is perfectly done
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Davidosky007 reviewed (changes requested):
Hi @ITurres 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well written PR 👍 
* Good looking readme 👍 
* Multiple tables queried 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ITurres committed:
fix: on 'schema.sql' Remove/Add columns

- Remove hardcoded foreign keys references and its columns from
  -animals- table.

- Set 'species' column back.
- Use ALTER to:
    - DROP the 'species' column.
    - ADD the 'species_id' and
      'owner_id' columns with their foreign keys references.

nahnahdev reviewed (approved):
## Project Status: APPROVED :green_circle: 
Hi @ITurres , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
<img src="https://media3.giphy.com/media/n0eJLge10ujSM/giphy.gif"/>

## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ITurres commented:
Hi @NahnahAJ! thank you for reviewing this PR too! :D I want to thank you all @bobb-Rob && @Davidosky007 for providing insightful feedback and pointing me in the right direction! 🎯 it is always very appreciated.

I wish you all a good week and happy coding! 🪂
